api/v2/pinning.go

```diff
@@ -0,0 +1,172 @@
+package api
+
+import (
+	"fmt"
+	"github.com/application-research/estuary/api/v1"
+	"github.com/application-research/estuary/collections"
+	"github.com/application-research/estuary/pinner/types"
+	"github.com/application-research/estuary/util"
+	"github.com/ipfs/go-cid"
+	"github.com/labstack/echo/v4"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/xerrors"
+	"gorm.io/gorm"
+	"net/http"
+	"path/filepath"
+)
+
+type BatchedPinRequest struct {
+	ContentIdToPin string `json:"content_id"`
+}
+
+// handleGetBatchedPins  godoc
+// @Summary      Get a pin status object
+// @Description  This endpoint returns a pin status object.
+// @Tags         pinning
+// @Produce      json
+// @Success      200	{object}  []types.IpfsPinStatusResponse
+// @Failure      404	{object}  util.HttpError
+// @Failure      500    {object}  util.HttpError
+// @Param        pin           body      []api.BatchedPinRequest  true   "Pin Body {[content_id:"content_id_to_pin"]}"
+// @Router       /v2/pinning/batched-pins/ [get]
+func (s *apiV2) handleGetBatchedPins(c echo.Context, u *util.User) error {
+
+	var pins []BatchedPinRequest
+	err := c.Bind(&pins)
+	if err != nil {
+		return err
+	}
+
+	var pinStatuses []*types.IpfsPinStatusResponse
+	for _, pinId := range pins {
```

Comment:
nit: change `pinId` to `pin`

Reply:
also, can we abstract these lines below and reuse them for both single and batch

Reply:
@anjor do you mean the batch endpoint should make an HTTP call to the single endpoint? If no, then we are saying the same thing, if yes, I do not think that is a good approach

Reply:
I think he means having a helper function that both /pinning/pins and /pinning/batched-pins call, only that batched-pins will call it several times.

Reply:
> also, can we abstract these lines below and reuse them for both single and batch

that is what I meant by this

Reply:
Yeah, except having multiple individual helper functions, just having a single one (which could then call separate individual helper functions).

so if the pinning endpoint is 
```
doPin()
```

and the batch pinnning endpoint is
```
for {
 doPin()
}
```


instead of pinning endpoint being
```
doOneThing()
doAnotherThing()
doYetAnotherThing()
```
and batch endpoint being
```
for {
doOneThing()
doAnotherThing()
doYetAnotherThing()

}
```